### PR TITLE
[OSCD] Modified "WMI Persistence" Rule II

### PR DESCRIPTION
**Pull request by GlebSukhodolskiy**
PR #1161 

Issue #576 Task 71

### CODE REVIEW

**Comment by GlebSukhodolskiy** [comment]
@aw350m33d Your suggestion from PR #1161 was implemented here :)

File: rules/windows/other/win_wmi_persistence.yml [hunk 1]
```diff
@@ -1,33 +1,52 @@
...
     product: windows
-    service: wmi
+    service: wmi #native windows detection
+    definition: 'WMI Namespaces Auditing and SACL should be configured, EventID 5861 and 5859 detection requires Windows 10, 2012 and higher'
 detection:
-    selection:
+    wmi_filter_to_consumer_binding:
         EventID: 5861
     keywords:
         Message:
```
Comment (aw350m33d) [inline rules/windows/other/win_wmi_persistence.yml:23]: Let's use field modifier `contains` here. We try to use field modifiers wherever possible. 😃

**Review by aw350m33d — CHANGES_REQUESTED** [review]

File: rules/windows/other/win_wmi_persistence.yml [hunk 2]
```diff
@@ -1,33 +1,52 @@
...
+detection:
+    network_logon:
+        EventID: 4624
+        LogonType: 3
+    privileges_assigned:
+        EventID: 4672
+    wmi_subscription:
+        EventID: 4662
+        ObjectType: 'WMI Namespace'
+        ObjectName: '*subscription*'
```
Comment (aw350m33d) [inline rules/windows/other/win_wmi_persistence.yml:51]: The same as for `keywords` items

**Review by aw350m33d — CHANGES_REQUESTED** [review]

File: rules/windows/other/win_wmi_persistence.yml [hunk 3]
```diff
@@ -1,33 +1,52 @@
+action: global
 title: WMI Persistence
 id: 0b7889b4-5577-4521-a60a-3376ee7f9f7b
 status: experimental
-description: Detects suspicious WMI event filter and command line event consumer based on event id 5861 and 5859 (Windows 10, 2012 and higher)
-author: Florian Roth
+description: Detects suspicious WMI event filter and command line event consumer based on WMI and Sysmon logs.
```
Comment (aw350m33d) [inline rules/windows/other/win_wmi_persistence.yml:5]: Since the part with sysmon events was excluded, the description needs to be fixed.

**Review by aw350m33d — CHANGES_REQUESTED** [review]

File: rules/windows/other/win_wmi_persistence.yml (hunk 3 above)
Comment (GlebSukhodolskiy) [reply rules/windows/other/win_wmi_persistence.yml]: Done

File: rules/windows/other/win_wmi_persistence.yml [hunk 4]
```diff
@@ -1,33 +1,51 @@
...
-author: Florian Roth
+description: Detects suspicious WMI event filter and command line event consumer based on WMI and Security Logs.
+author: Florian Roth, Gleb Sukhodolskiy, Timur Zinniatullin oscd.community
 date: 2017/08/22
-modified: 2020/08/23
+modified: 2020/10/13
 references:
     - https://twitter.com/mattifestation/status/899646620148539397
     - https://www.eideon.com/2018-03-02-THL03-WMIBackdoors/
+    - https://attack.mitre.org/techniques/T1546/003/
```
Comment (yugoslavskiy) [inline rules/windows/other/win_wmi_persistence.yml:12]: Hello @GlebSukhodolskiy! could you please remove the ATT&CK tag from references, as the technique has already been mentioned in the `tags`?

Comment (GlebSukhodolskiy) [reply rules/windows/other/win_wmi_persistence.yml]: Sure, done :)

Comment (yugoslavskiy) [reply rules/windows/other/win_wmi_persistence.yml]: Thanks!

**Comment by yugoslavskiy** [comment]
Thank you very much, @GlebSukhodolskiy! Well done!

**Review by yugoslavskiy — APPROVED** [review]
@thomaspatzke @Neo23x0 guys, I believe that this PR could be merged (:
please let us know what do you think about it if there is anything you would like to change.

**Comment by yugoslavskiy** [comment]
Hello @GlebSukhodolskiy, @zinint, guys! There are conflicts in the Rule. Could you please take a look and resolve them?
BTW, the following detection logic:

```
detection:
    network_logon:
        EventID: 4624
        LogonType: 3
    privileges_assigned:
        EventID: 4672
    wmi_subscription:
        EventID: 4662
        ObjectType: 'WMI Namespace'
        ObjectName|contains: 'subscription'

```

with the following condition:

```
    condition: network_logon and privileges_assigned and wmi_subscription
```

Is not going to fly as there are different EventIDs with `AND` condition.

**Comment by GlebSukhodolskiy** [comment]
> Hello @GlebSukhodolskiy, @zinint, guys! There are conflicts in the Rule. Could you please take a look and resolve them?
> BTW, the following detection logic:
> Is not going to fly as there are different EventIDs with `AND` condition.

Hello! @yugoslavskiy I understand what is a problem, but have no clue how to resolve it. The idea behind the rule is that it looks for a sequence of events. First 4624 -> 4672 -> 4662. Is there any way in Sigma to specify exactly sequences? I didn't find anything similar in the spec.

Otherwise, the only option is to delete the first two EIDs. What do you think?

**Comment by Neo23x0** [comment]
No, see: https://twitter.com/cyb3rops/status/1338180505511063554

**Comment by yugoslavskiy** [comment]
Thank you for your contribution, @GlebSukhodolskiy!

**Comment by yugoslavskiy** [comment]
@GlebSukhodolskiy could you please resolve the conflicts so I could merge the PR?

**Comment by GlebSukhodolskiy** [comment]
> @GlebSukhodolskiy could you please resolve the conflicts so I could merge the PR?

Done!

**Pull request merged**